internal/controllers/cluster/cluster_controller.go

```diff
@@ -455,7 +455,7 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, s *scope) (reconcile.R
 			s.deletingMessage = fmt.Sprintf("Waiting for %s to be deleted", cluster.Spec.ControlPlaneRef.Kind)
 
 			log.Info("Cluster still has descendants - need to requeue", "controlPlaneRef", cluster.Spec.ControlPlaneRef.Name)
-			return ctrl.Result{}, nil
+			return ctrl.Result{RequeueAfter: deleteRequeueAfter}, nil
```

Comment:
I will prefer to not use requeAfter unless we have clear evidence that we are missing some events, and exploring first if we can improve our watches.

If the issue is the message, we can improve it by replacing "need to requeue" with "waiting for object deletion" or something similar